storage/cloud-client/bucket_lock.py

```diff
@@ -0,0 +1,213 @@
+    # [END storage_disable_default_event_based_hold]
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    subparsers = parser.add_subparsers(dest='command')
+
+    set_retention_policy_parser = subparsers.add_parser(
+        'set-retention-policy', help=set_retention_policy.__doc__)
+    set_retention_policy_parser.add_argument('bucket_name')
+    set_retention_policy_parser.add_argument('retention_period')
+
+    remove_retention_policy_parser = subparsers.add_parser(
+        'remove-retention-policy', help=remove_retention_policy.__doc__)
+    remove_retention_policy_parser.add_argument('bucket_name')
+    remove_retention_policy_parser.add_argument('retention_period')
+
+    lock_retention_policy_parser = subparsers.add_parser(
+        'lock-retention-policy', help=lock_retention_policy.__doc__)
+    lock_retention_policy_parser.add_argument('bucket_name')
+
+    set_temporary_hold_parser = subparsers.add_parser(
+        'set-temporary-hold', help=set_temporary_hold.__doc__)
+    set_temporary_hold_parser.add_argument('bucket_name')
+    set_temporary_hold_parser.add_argument('retention_period')
+
+    release_temporary_hold_parser = subparsers.add_parser(
+        'release-temporary-hold', help=release_temporary_hold.__doc__)
+    release_temporary_hold_parser.add_argument('bucket_name')
+    release_temporary_hold_parser.add_argument('blob_name')
+
+    set_event_based_hold_parser = subparsers.add_parser(
+        'set-event_based-hold', help=set_event_based_hold.__doc__)
+    set_event_based_hold_parser.add_argument('bucket_name')
+    set_event_based_hold_parser.add_argument('blob_name')
+
+    release_event_based_hold_parser = subparsers.add_parser(
+        'release-event_based-hold', help=release_event_based_hold.__doc__)
+    release_event_based_hold_parser.add_argument('bucket_name')
+    release_event_based_hold_parser.add_argument('blob_name')
+
+    enable_default_event_based_hold_parser = subparsers.add_parser(
+        'enable-default-event-based-hold',
+        help=enable_default_event_based_hold.__doc__)
+    enable_default_event_based_hold_parser.add_argument('bucket_name')
+
+    disable_default_event_based_hold_parser = subparsers.add_parser(
+        'disable-default-event-based-hold',
+        help=disable_default_event_based_hold.__doc__)
+    disable_default_event_based_hold_parser.add_argument('bucket_name')
+
+    args = parser.parse_args()
+
+    if args.command == 'set-retention-policy':
+        set_retention_policy(args.bucket_name, args.retention_period)
+    elif args.command == 'remove-retention-policy':
+        remove_retention_policy(args.bucket_name)
+    elif args.command == 'lock-retention-policy':
+        lock_retention_policy(args.bucket_name)
+    elif args.command == 'set-temporary-hold':
+        set_temporary_hold(args.bucket_name)
```

Comment:
missing blob_name, same for: `release_temporary_hold`, `set_event_based_hold`, `release_event_based_hold`

Reply:
Done